### PR TITLE
add build.yaml

### PR DESCRIPTION
**Pull request by airycanon**
Thank you for contributing to Harbor!

# Comprehensive Summary of your change

# Issue being fixed
Fixes #(issue)

Please indicate you've done the following:
- [ ] Well Written Title and Summary of the PR
- [ ] Label the PR as needed. "release-note/ignore-for-release, release-note/new-feature, release-note/update, release-note/enhancement, release-note/community, release-note/breaking-change, release-note/docs, release-note/infra, release-note/deprecation"
- [ ] Accepted the DCO. Commits without the DCO will delay acceptance.
- [ ] Made sure tests are passing and test coverage is added if needed.
- [ ] Considered the docs impact and opened a new docs issue or PR with docs changes if needed in [website repository](https://github.com/goharbor/website).

### CODE REVIEW

File: make/photon/nginx/Dockerfile.base [hunk 1]
```diff
@@ -1,7 +1,7 @@
 FROM photon:2.0
 
-RUN tdnf install -y nginx shadow >> /dev/null \
+RUN groupadd -r -g 10000 nginx && useradd --no-log-init -r -g 10000 -u 10000 nginx \
```
Comment (airycanon) [inline make/photon/nginx/Dockerfile.base:3]: 先创建 group 和  user 是因为安装 nginx 时也会创建，但 id 不是 10000

File: make/photon/portal/Dockerfile.base [hunk 2]
```diff
@@ -1,8 +1,8 @@
 FROM photon:2.0
 
-RUN tdnf install -y nginx shadow >> /dev/null \
+RUN groupadd -r -g 10000 nginx && useradd --no-log-init -r -g 10000 -u 10000 nginx \
```
Comment (airycanon) [inline make/photon/portal/Dockerfile.base:3]: 同上

File: make/photon/registry/builder [hunk 3]
```diff
@@ -25,13 +25,13 @@ git clone -b $VERSION https://github.com/distribution/distribution.git $TEMP
 # add patch 2879
 echo 'add patch https://github.com/distribution/distribution/pull/2879 ...'
 cd $TEMP
-wget https://github.com/distribution/distribution/pull/2879.patch
+curl -O -L https://github.com/distribution/distribution/pull/2879.patch
```
Comment (airycanon) [inline make/photon/registry/builder:28]: 基础镜像中没有 wget

File: make/photon/registry/builder [hunk 4]
```diff
@@ -25,13 +25,13 @@ git clone -b $VERSION https://github.com/distribution/distribution.git $TEMP
...
 git apply 2879.patch
 
 # add patch 3370
 echo 'add patch https://github.com/distribution/distribution/pull/3370 ...'
 cd $TEMP
-wget https://github.com/distribution/distribution/pull/3370.patch
+curl -O -L https://github.com/distribution/distribution/pull/3370.patch
```
Comment (airycanon) [inline make/photon/registry/builder:34]: 同上

File: make/patches/patch-arm64.sh [hunk 5]
```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
```
Comment (airycanon) [inline make/patches/patch-arm64.sh:1]: 从 images-sync 仓库拷贝过来的。

File: make/patches/patch-amd64.sh [hunk 6]
```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
```
Comment (airycanon) [inline make/patches/patch-amd64.sh:1]: 从 images-sync 仓库拷贝过来的。

File: .build/build.yaml [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+    - name: build-image
+      runAfter:
+        - generate-version
+      workspaces:
+        - name: source
+          workspace: source
+      params:
+        - name: version
+          value: 2.2.3-$(tasks.generate-version.results.commit-short-id)
+        - name: container-images
```
Comment (airycanon) [inline .build/build.yaml:49]: 这个参数没有必要，加了是为了让 build-harbor 的凭据可以同步到 task 中。

**Review by nanjingfm — APPROVED** [review]

**Pull request merged**